src/python/pants/option/arg_splitter.py

```diff
@@ -201,6 +202,16 @@ def assign_flag_to_scope(flg: str, default_scope: str) -> None:
         if not goals and not self._help_request:
             self._help_request = NoGoalHelp()
 
+        for goal in goals:
+            si = self._known_goal_scopes[goal]
+            if si.deprecated_scope and goal == si.deprecated_scope:
+                assert si.deprecated_scope_removal_version
```

Comment:
What is the benefit of this assert? If it fires, the user will get a not-very-helpful error message. And we already validate that argument in warn_or_error.

Reply:
Yea, but mypy does not know that if `deprecated_scope` is set, then `deprecated_scope_removal_version` must also be. I could defensively check here, but this would probably be the wrong place for that check anyway.